File: src/main/java/tech/bobliu/assignment08/model/Post.java
package tech.bobliu.assignment08.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Table(name = "posts")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User sender;

    @Column(columnDefinition = "TEXT")
    private String content;

    private Date datetime = new Date();

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reply_to")
    private Post parent;

    @ManyToOne
    @JoinColumn(name = "thread_id")
    private Thread thread;
}

File: src/main/kotlin/tech/bobliu/assignment08/model/Post.kt
package tech.bobliu.assignment08.model

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "posts")
class Post(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    var sender: User? = null,

    @Column(columnDefinition = "TEXT")
    var content: String = "",

    var datetime: Date = Date(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reply_to")
    var parent: Post? = null,

    @ManyToOne
    @JoinColumn(name = "thread_id")
    var thread: Thread? = null,
)
